std/uni.d

```diff
@@ -7094,11 +7103,24 @@ int icmp(S1, S2)(S1 str1, S2 str2)
 }
 
 ///
-unittest{
+unittest
+{
     assert(icmp("Rußland", "Russland") == 0);
     assert(icmp("ᾩ -> \u1F70\u03B9", "\u1F61\u03B9 -> ᾲ") == 0);
 }
 
+/**
+ * By using $(XREF utf, byUTF) and its aliases, GC allocations via auto-decoding
+ * and thrown exceptions can be avoided, making `icmp` `@safe @nogc nothrow pure`.
+ */
+@safe @nogc nothrow pure unittest
+{
+    import std.utf : byDchar;
+
+    assert(icmp("Rußland".byDchar, "Russland".byDchar) == 0);
```

Comment:
Wait what??? 
In German (my mother tongue) this is not guaranteed to be correct. From Wikipedia:

Thus it helps to distinguish words like Buße (long vowel) 'penance, fine' and Busse (short vowel) 'buses'. 

https://en.m.wikipedia.org/wiki/%C3%9F